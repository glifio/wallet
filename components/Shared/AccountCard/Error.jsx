import React from 'react'
import { string, func } from 'prop-types'
import Box from '../Box'
import Glyph from '../Glyph'
import Button from '../Button'
import { Text } from '../Typography'

const AccountError = ({ errorMsg, onTryAgain }) => (
  <Box
    display='flex'
    flexDirection='column'
    justifyContent='space-between'
    width={11}
    height={11}
    borderRadius={2}
    p={3}
    bg='card.error.background'
    color='card.error.foreground'
    boxShadow={1}
    m={2}
  >
    <Box display='flex' alignItems='center' justifyContent='flex-start'>
      <Glyph mr={3} acronym='Er' color='status.fail.foreground' />
      <Text>Error</Text>
    </Box>
    <Box>
      <Text margin={0}>{errorMsg}</Text>
    </Box>
    <Box display='flex'>
      <Button
        title='Try again'
        onClick={onTryAgain}
        height='100%'
        backgroundColor='core.transparent'
        borderColor='status.fail.foreground'
        color='status.fail.foreground'
        p={2}
      />
    </Box>
  </Box>
)

AccountError.propTypes = {
  onTryAgain: func.isRequired,
  errorMsg: string
}

AccountError.defaultProps = {
  errorMsg: ''
}

export default AccountError
