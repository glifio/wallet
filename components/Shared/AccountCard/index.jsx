import React, { forwardRef } from 'react'
import { string, func, bool, oneOf } from 'prop-types'
import { ADDRESS_PROPTYPE } from '../../../customPropTypes'
import Box from '../Box'
import Glyph from '../Glyph'
import Button from '../Button'
import { CopyAddress } from '../Copy'
import { BigTitle, Text } from '../Typography'
import {
  LEDGER,
  CREATE_MNEMONIC,
  IMPORT_MNEMONIC,
  IMPORT_SINGLE_KEY
} from '../../../constants'

const AccountCard = forwardRef(
  (
    {
      address,
      alias,
      onAccountSwitch,
      onShowOnLedger,
      ledgerBusy,
      walletType,
      ...props
    },
    ref
  ) => {
    return (
      <Box
        display='flex'
        flexDirection='column'
        justifyContent='space-between'
        width='300px'
        height='300px'
        borderRadius={3}
        p={3}
        color='card.account.color'
        bg='card.account.background'
        boxShadow={1}
        ref={ref}
        {...props}
      >
        <Box
          display='flex'
          alignItems='center'
          justifyContent='flex-start'
          color='card.account.color'
        >
          <Glyph mr={3} color='card.account.color' acronym='Ac' />
          <Text>Account</Text>
        </Box>
        <Box color='card.account.color'>
          <BigTitle>{alias}</BigTitle>
          <CopyAddress address={address} />
        </Box>
        <Box display='flex'>
          {walletType !== IMPORT_SINGLE_KEY && (
            <Button
              variant='tertiary'
              title='Switch'
              onClick={onAccountSwitch}
              height='max-content'
              p={2}
              py={2}
            />
          )}
          {walletType === LEDGER && (
            <Button
              variant='tertiary'
              title={ledgerBusy ? 'Check Device' : 'Show on Device'}
              onClick={onShowOnLedger}
              height='max-content'
              ml={2}
              p={2}
              py={2}
              disabled={ledgerBusy}
            />
          )}
        </Box>
      </Box>
    )
  }
)

AccountCard.propTypes = {
  /**
   * Filecoin address
   */
  address: ADDRESS_PROPTYPE,
  /**
   * Human readable alias of Filecoin address
   */
  alias: string.isRequired,
  /**
   * Sets background-color of the card
   */
  color: string,
  /**
   * Fired when the "switch" button is clicked
   */
  onAccountSwitch: func.isRequired,
  /**
   * If this wallet represents a ledger
   */
  walletType: oneOf([
    LEDGER,
    CREATE_MNEMONIC,
    IMPORT_MNEMONIC,
    IMPORT_SINGLE_KEY
  ]).isRequired,
  /**
   * If this wallet represents a ledger, the function that gets called when "show on Ledger" button gets clicked
   */
  onShowOnLedger: func,
  /**
   * When true, disable the show on ledger button
   */
  ledgerBusy: bool
}

AccountCard.defaultProps = {
  color: 'white',
  onShowOnLedger: () => {},
  ledgerBusy: false
}

export default AccountCard
