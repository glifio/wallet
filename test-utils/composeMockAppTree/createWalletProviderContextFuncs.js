import { FilecoinNumber } from '@glif/filecoin-number'

import {
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../../WalletProvider/state'

export default walletProviderDispatch => ({
  fetchDefaultWallet: jest.fn().mockImplementation(() => ({
    balance: new FilecoinNumber('1', 'fil'),
    address: 't1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq',
    path: "m/44'/1'/0/0/0"
  })),
  setWalletError: jest
    .fn()
    .mockImplementation(errorMessage =>
      walletProviderDispatch(setError(errorMessage))
    ),
  setWalletType: jest
    .fn()
    .mockImplementation(walletType =>
      walletProviderDispatch(setWalletType(walletType))
    ),
  setLedgerProvider: jest.fn(),
  connectLedger: jest.fn(),
  resetLedgerState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetLedgerState())
  }),
  resetState: jest.fn().mockImplementation(() => {
    walletProviderDispatch(resetState())
  })
})
