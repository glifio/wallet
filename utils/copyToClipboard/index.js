const fallbackCopyTextToClipboard = text => {
  const textArea = document.createElement('textarea')
  textArea.value = text
  textArea.style.position = 'fixed' // avoid scrolling to bottom
  document.body.appendChild(textArea)
  textArea.focus()
  textArea.select()
  document.execCommand('copy')
  document.body.removeChild(textArea)
}

export default text =>
  new Promise((resolve, reject) => {
    if (!navigator.clipboard) {
      fallbackCopyTextToClipboard(text)
      return resolve()
    }
    navigator.clipboard.writeText(text).then(resolve, reject)
  })
