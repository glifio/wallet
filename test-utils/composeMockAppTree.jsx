/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import cloneDeep from 'lodash.clonedeep'

import WalletProvider from '@openworklabs/filecoin-wallet-provider'
import { FilecoinNumber } from '@openworklabs/filecoin-number'
import { Provider } from 'react-redux'

import { NetworkCheck } from '../lib/check-network'
import BalancePoller from '../lib/update-balance'
import { ConverterContext } from '../lib/Converter'
import { theme, ThemeProvider } from '../components/Shared'
import { initializeStore } from './index'
import { initialState } from '../store/states'
import walletProviderReducer, {
  initialState as walletProviderInitialState,
  setWalletType,
  setError,
  resetLedgerState,
  resetState
} from '../WalletProvider/state'
import { WalletProviderContext } from '../WalletProvider'
import createPath from '../utils/createPath'
import { IMPORT_MNEMONIC } from '../constants'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

const mockRouter = jest.fn(() => {})
useRouter.mockImplementation(() => ({
  query: 'network=t',
  push: mockRouter,
  replace: mockRouter
}))

jest.mock('@openworklabs/filecoin-wallet-provider')
const mockGetAccounts = jest
  .fn()
  .mockImplementation(() =>
    Promise.resolve(['t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'])
  )

const mockGetBalance = jest
  .fn()
  .mockImplementation(() => Promise.resolve(new FilecoinNumber('1', 'fil')))

WalletProvider.mockImplementation(() => {
  return {
    wallet: {
      getAccounts: mockGetAccounts,
      sign: jest.fn().mockImplementation(() => 'xxxyyyyzzzz')
    },
    getBalance: mockGetBalance,
    getNonce: jest.fn().mockImplementation(() => 0),
    estimateGas: jest
      .fn()
      .mockImplementation(() => new FilecoinNumber('122', 'attofil')),
    sendMessage: jest.fn().mockImplementation(() => 'QmZCid!')
  }
})

const mockWalletProviderInstance = new WalletProvider()

const presets = {
  preOnboard: cloneDeep(initialState),
  postOnboard: cloneDeep({
    ...initialState,
    wallets: [
      {
        address: 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi',
        balance: new FilecoinNumber('1', 'fil'),
        path: createPath(1, 0)
      }
    ],
    selectedWalletIdx: 0
  })
}

const composeWalletProviderState = (initialWalletProviderState, preset) => {
  switch (preset) {
    case 'postOnboard': {
      return Object.freeze({
        ...initialWalletProviderState,
        walletType: IMPORT_MNEMONIC,
        walletProvider: mockWalletProviderInstance
      })
    }
    default:
      return initialWalletProviderState
  }
}

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const mockConverterInstance = options.mockConverterInstance || {
    cacheConversionRate: jest.fn(),
    toFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').dividedBy(5)
    }),
    fromFIL: jest.fn().mockImplementation(amount => {
      return new FilecoinNumber(amount, 'fil').multipliedBy(5)
    })
  }

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )
    return (
      <Provider store={store}>
        <ConverterContext.Provider
          value={{
            converter: mockConverterInstance,
            converterError: options.converterError || null
          }}
        >
          <WalletProviderContext.Provider
            value={{
              state: walletProviderState,
              dispatch: walletProviderDispatch,
              fetchDefaultWallet: jest.fn().mockImplementation(),
              setWalletError: jest
                .fn()
                .mockImplementation(errorMessage =>
                  walletProviderDispatch(setError(errorMessage))
                ),
              setWalletType: jest
                .fn()
                .mockImplementation(walletType =>
                  walletProviderDispatch(setWalletType(walletType))
                ),
              setLedgerProvider: jest.fn(),
              connectLedger: jest.fn(),
              resetLedgerState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetLedgerState())
              }),
              resetState: jest.fn().mockImplementation(() => {
                walletProviderDispatch(resetState())
              })
            }}
          >
            <NetworkCheck
              networkFromRdx={store.getState().network}
              pathname={pathname}
              query={query}
              switchNetwork={jest.fn()}
            />
            <BalancePoller />
            <ThemeProvider theme={theme}>{children}</ThemeProvider>
          </WalletProviderContext.Provider>
        </ConverterContext.Provider>
      </Provider>
    )
  }

  return { Tree, store, walletProvider: mockWalletProviderInstance }
}
