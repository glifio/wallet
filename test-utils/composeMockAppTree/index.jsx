/* eslint-disable react/prop-types */
import React, { useReducer } from 'react'
import { Provider } from 'react-redux'
import { Converter } from '@openworklabs/filecoin-number'

import { NetworkCheck } from '../../lib/check-network'
import BalancePoller from '../../lib/update-balance'
import { ConverterContext } from '../../lib/Converter'
import { theme, ThemeProvider } from '../../components/Shared'
import { initializeStore } from '..'
import walletProviderReducer, {
  initialState as walletProviderInitialState
} from '../../WalletProvider/state'
import { WasmContext } from '../../lib/WasmLoader'
import { WalletProviderContext } from '../../WalletProvider'
import { mockWalletProviderInstance } from '../mocks/mock-wallet-provider'
import createMockWalletProviderContextFuncs from './createWalletProviderContextFuncs.js'
import mockWalletSubproviders from '../mocks/mock-wallet-subproviders'
import { presets, composeWalletProviderState } from './composeState'

/**
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * walletProviderDispatch(a dispatcher to mock/read calls to the walletProvider's state)
 * pathname
 * query (usually { network: 'f' } or { network: 't' })
 */

export default (statePreset = 'preOnboard', options = {}) => {
  const state = options.state || presets[statePreset]
  const store = initializeStore(state)

  const pathname = options.pathname || '/wallet'
  const query = options.query || { network: 't' }

  const Tree = ({ children }) => {
    const [initialWalletProviderState, walletProviderDispatch] = useReducer(
      options.reducer || walletProviderReducer,
      options.walletProviderInitialState || walletProviderInitialState
    )

    const walletProviderState = composeWalletProviderState(
      initialWalletProviderState,
      statePreset
    )

    const mockWalletProviderContextFuncs = createMockWalletProviderContextFuncs(
      options.walletProviderDispatch || walletProviderDispatch
    )

    return (
      <Provider store={store}>
        <WasmContext.Provider
          value={{
            generateMnemonic: jest
              .fn()
              .mockImplementation(
                () =>
                  'slender spread awkward chicken noise useful thank dentist tip bronze ritual explain version spot collect whisper glow peanut bus local country album punch frown'
              )
          }}
        >
          <ConverterContext.Provider
            value={{
              converter: new Converter(),
              converterError: options.converterError || null
            }}
          >
            <WalletProviderContext.Provider
              value={{
                state: walletProviderState,
                dispatch:
                  options.walletProviderDispatch || walletProviderDispatch,
                walletSubproviders: {
                  ...mockWalletSubproviders
                },
                ...mockWalletProviderContextFuncs
              }}
            >
              <NetworkCheck
                networkFromRdx={store.getState().network}
                pathname={pathname}
                query={query}
                switchNetwork={jest.fn()}
              />
              <BalancePoller />
              <ThemeProvider theme={theme}>{children}</ThemeProvider>
            </WalletProviderContext.Provider>
          </ConverterContext.Provider>
        </WasmContext.Provider>
      </Provider>
    )
  }

  return {
    Tree,
    store,
    walletProvider: mockWalletProviderInstance
  }
}
