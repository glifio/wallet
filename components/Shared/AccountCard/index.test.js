import { cleanup, render, screen, act, fireEvent } from '@testing-library/react'
import AccountCard from '.'
import composeMockAppTree from '../../../test-utils/composeMockAppTree'
jest.mock('../../../utils/copyToClipboard')
import copyToClipboard from '../../../utils/copyToClipboard'

describe('AccountCard', () => {
  afterEach(cleanup)
  test('renders the card', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const { container } = render(
      <Tree>
        <AccountCard
          onAccountSwitch={() => {}}
          color='purple'
          alias='Prime'
          address={'t0123456789'}
          walletType={'LEDGER'}
          onShowOnLedger={() => {}}
          ledgerBusy={false}
          mb={2}
        />
      </Tree>
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the card CREATE_MNEMONIC', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const { container } = render(
      <Tree>
        <AccountCard
          onAccountSwitch={() => {}}
          color='purple'
          alias='Prime'
          address={'t0123456789'}
          walletType={'CREATE_MNEMONIC'}
          onShowOnLedger={() => {}}
          ledgerBusy={false}
          mb={2}
        />
      </Tree>
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders alias and address', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    render(
      <Tree>
        <AccountCard
          onAccountSwitch={() => {}}
          color='purple'
          alias='Prime'
          address='t0123'
          walletType='LEDGER'
          onShowOnLedger={() => {}}
          ledgerBusy={false}
          mb={2}
        />
      </Tree>
    )

    expect(screen.getByText('t0123', { exact: false })).toBeInTheDocument()
    expect(screen.getByText('Prime')).toBeInTheDocument()
  })

  test('clicking "Switch" calls onAccountSwitch', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const mockOnAccountSwitch = jest.fn()
    const { getByText } = render(
      <Tree>
        <AccountCard
          onAccountSwitch={mockOnAccountSwitch}
          color='purple'
          alias='Prime'
          address='t0123'
          walletType='LEDGER'
          onShowOnLedger={() => {}}
          ledgerBusy={false}
          mb={2}
        />
      </Tree>
    )

    act(() => {
      fireEvent.click(getByText('Switch'))
    })

    expect(mockOnAccountSwitch).toHaveBeenCalled()
  })

  test('clicking "Show on Device" calls onShowOnLedger', () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const mockOnShowOnLedger = jest.fn()
    const { getByText } = render(
      <Tree>
        <AccountCard
          onAccountSwitch={() => {}}
          color='purple'
          alias='Prime'
          address='t0123'
          walletType='LEDGER'
          onShowOnLedger={mockOnShowOnLedger}
          ledgerBusy={false}
          mb={2}
        />
      </Tree>
    )

    act(() => {
      fireEvent.click(getByText('Show on Device'))
    })

    expect(mockOnShowOnLedger).toHaveBeenCalled()
  })

  test('clicking "Copy" calls copy', async () => {
    const { Tree } = composeMockAppTree('postOnboard')
    const mockCopyToClipboard = jest.fn(() => Promise.resolve('yo'))
    copyToClipboard.mockImplementationOnce(mockCopyToClipboard)

    const { getByText } = render(
      <Tree>
        <AccountCard
          onAccountSwitch={() => {}}
          color='purple'
          alias='Prime'
          address='t0123'
          walletType='LEDGER'
          onShowOnLedger={() => {}}
          ledgerBusy={false}
          mb={2}
        />
      </Tree>
    )

    act(() => {
      fireEvent.click(getByText('Copy'))
    })

    expect(mockCopyToClipboard).toHaveBeenCalled()
  })
})
