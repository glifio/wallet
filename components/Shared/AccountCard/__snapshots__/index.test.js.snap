// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCard renders the card 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  border-radius: 8px;
  padding: 16px;
  margin-bottom: 8px;
  color: purple;
  background-color: #5E26FF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 300px;
  height: 300px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  box-shadow: 0 0.7px 2.2px -8px rgba(0,0,0,0.18),0 1.7px 5.4px rgba(0,0,0,0.129),0 3.1px 10.1px rgba(0,0,0,0.107),0 5.6px 18.1px rgba(0,0,0,0.09),0 10.4px 33.8px rgba(0,0,0,0.073),0 25px 81px rgba(0,0,0,0.051);
}

.c1 {
  min-width: 0;
  box-sizing: border-box;
  color: #E0D7FE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-radius: 8px;
  border-style: solid;
  margin-right: 16px;
  color: #E0D7FE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  min-width: 0;
  box-sizing: border-box;
  color: #E0D7FE;
}

.c7 {
  min-width: 0;
  box-sizing: border-box;
  color: #E0D7FE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

.c17 {
  cursor: pointer;
  border: 1px solid #E0D7FE;
  background-color: transparent;
  border-color: #E0D7FE;
  color: #E0D7FE;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  padding: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  border-width: 1px;
  border-radius: 4px;
}

.c17:hover {
  opacity: 0.8;
}

.c18 {
  cursor: pointer;
  border: 1px solid #E0D7FE;
  background-color: transparent;
  border-color: #E0D7FE;
  color: #E0D7FE;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-left: 8px;
  padding: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  border-width: 1px;
  border-radius: 4px;
}

.c18:hover {
  opacity: 0.8;
}

.c12 {
  width: 24;
  height: 24;
}

.c8 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.2;
  font-family: RT-Alias-Grotesk;
  margin: 0;
}

.c4 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c6 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin: 0;
}

.c14 {
  color: #E0D7FE;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1;
  font-family: RT-Alias-Grotesk;
  text-align: left;
  min-width: 64px;
  margin: 0;
  margin-top: 0;
  margin-left: 4px;
}

.c9 {
  white-space: nowrap;
}

.c11 {
  cursor: pointer;
  border: 1px solid #1AD08F;
  background-color: #1AD08F;
  border-color: #1AD08F;
  color: #08442F;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-left: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-width: 1px;
  border-radius: 4px;
  height: auto;
  opacity: 1 !important;
  border: 0;
  background: transparent;
  padding: 0;
  outline: none;
}

.c11:hover {
  opacity: 0.8;
}

.c13 {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c10:hover .c13 {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c15 {
  -webkit-transition: 0.18s ease-in;
  transition: 0.18s ease-in;
  opacity: 0;
}

.c10:hover .c15 {
  opacity: 1;
}

<div
  class="c0"
  color="purple"
  display="flex"
  height="300px"
  width="300px"
>
  <div
    class="c1"
    color="card.account.color"
    display="flex"
  >
    <div
      class="c2"
      color="card.account.color"
      display="flex"
      size="6"
    >
      <h3
        class="c3"
      >
        Ac
      </h3>
    </div>
    <p
      class="c4"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
    >
      Account
    </p>
  </div>
  <div
    class="c5"
    color="card.account.color"
  >
    <p
      class="c6"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
    >
      Your Address
    </p>
    <div
      class="c7"
      color="core.secondary"
      display="flex"
    >
      <h2
        class="c8 c9"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="1"
      >
        t01234 ... 456789
      </h2>
      <button
        class="c10 c11"
        display="flex"
        font-size="3"
        height="6"
      >
        <svg
          class="c12 c13"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#1A0066"
            height="12"
            width="12"
            x="8"
            y="4"
          />
          <rect
            fill="#E0D7FE"
            height="12"
            width="12"
            x="4"
            y="8"
          />
        </svg>
        <h4
          class="c14 c15"
          color="core.secondary"
          font-family="RT-Alias-Grotesk"
          font-size="2"
          font-weight="400"
        >
          Copy
        </h4>
      </button>
    </div>
  </div>
  <div
    class="c16"
    display="flex"
  >
    <button
      class="c17"
      font-size="3"
      height="max-content"
      type="button"
    >
      Switch
    </button>
    <button
      class="c18"
      font-size="3"
      height="max-content"
      type="button"
    >
      Show on Device
    </button>
  </div>
</div>
`;

exports[`AccountCard renders the card CREATE_MNEMONIC 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  border-radius: 8px;
  padding: 16px;
  margin-bottom: 8px;
  color: purple;
  background-color: #5E26FF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 300px;
  height: 300px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  box-shadow: 0 0.7px 2.2px -8px rgba(0,0,0,0.18),0 1.7px 5.4px rgba(0,0,0,0.129),0 3.1px 10.1px rgba(0,0,0,0.107),0 5.6px 18.1px rgba(0,0,0,0.09),0 10.4px 33.8px rgba(0,0,0,0.073),0 25px 81px rgba(0,0,0,0.051);
}

.c1 {
  min-width: 0;
  box-sizing: border-box;
  color: #E0D7FE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-radius: 8px;
  border-style: solid;
  margin-right: 16px;
  color: #E0D7FE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  min-width: 0;
  box-sizing: border-box;
  color: #E0D7FE;
}

.c7 {
  min-width: 0;
  box-sizing: border-box;
  color: #E0D7FE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

.c17 {
  cursor: pointer;
  border: 1px solid #E0D7FE;
  background-color: transparent;
  border-color: #E0D7FE;
  color: #E0D7FE;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  padding: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  border-width: 1px;
  border-radius: 4px;
}

.c17:hover {
  opacity: 0.8;
}

.c12 {
  width: 24;
  height: 24;
}

.c8 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.2;
  font-family: RT-Alias-Grotesk;
  margin: 0;
}

.c4 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c6 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin: 0;
}

.c14 {
  color: #E0D7FE;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1;
  font-family: RT-Alias-Grotesk;
  text-align: left;
  min-width: 64px;
  margin: 0;
  margin-top: 0;
  margin-left: 4px;
}

.c9 {
  white-space: nowrap;
}

.c11 {
  cursor: pointer;
  border: 1px solid #1AD08F;
  background-color: #1AD08F;
  border-color: #1AD08F;
  color: #08442F;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-left: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-width: 1px;
  border-radius: 4px;
  height: auto;
  opacity: 1 !important;
  border: 0;
  background: transparent;
  padding: 0;
  outline: none;
}

.c11:hover {
  opacity: 0.8;
}

.c13 {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c10:hover .c13 {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c15 {
  -webkit-transition: 0.18s ease-in;
  transition: 0.18s ease-in;
  opacity: 0;
}

.c10:hover .c15 {
  opacity: 1;
}

<div
  class="c0"
  color="purple"
  display="flex"
  height="300px"
  width="300px"
>
  <div
    class="c1"
    color="card.account.color"
    display="flex"
  >
    <div
      class="c2"
      color="card.account.color"
      display="flex"
      size="6"
    >
      <h3
        class="c3"
      >
        Ac
      </h3>
    </div>
    <p
      class="c4"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
    >
      Account
    </p>
  </div>
  <div
    class="c5"
    color="card.account.color"
  >
    <p
      class="c6"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
    >
      Your Address
    </p>
    <div
      class="c7"
      color="core.secondary"
      display="flex"
    >
      <h2
        class="c8 c9"
        font-family="RT-Alias-Grotesk"
        font-size="3"
        font-weight="1"
      >
        t01234 ... 456789
      </h2>
      <button
        class="c10 c11"
        display="flex"
        font-size="3"
        height="6"
      >
        <svg
          class="c12 c13"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#1A0066"
            height="12"
            width="12"
            x="8"
            y="4"
          />
          <rect
            fill="#E0D7FE"
            height="12"
            width="12"
            x="4"
            y="8"
          />
        </svg>
        <h4
          class="c14 c15"
          color="core.secondary"
          font-family="RT-Alias-Grotesk"
          font-size="2"
          font-weight="400"
        >
          Copy
        </h4>
      </button>
    </div>
  </div>
  <div
    class="c16"
    display="flex"
  >
    <button
      class="c17"
      font-size="3"
      height="max-content"
      type="button"
    >
      Switch
    </button>
  </div>
</div>
`;
